pkg/aliauth/aliauth.go

```diff
@@ -2,132 +2,141 @@ package aliauth
 
 import (
 	"fmt"
-	"time"
+	"strings"
+	"unicode"
 
-	openapi "github.com/alibabacloud-go/darabonba-openapi/client"
-	"github.com/alibabacloud-go/sts-20150401/client"
+	openapi "github.com/alibabacloud-go/darabonba-openapi/v2/client"
 	"github.com/aliyun/aliyun-odps-go-sdk/odps/account"
-
-	openapiV2 "github.com/alibabacloud-go/darabonba-openapi/v2/client"
+	"github.com/aliyun/credentials-go/credentials"
+	"github.com/bearaujus/bptr"
+	"github.com/google/uuid"
 )
 
-var assumeRoleDefaultDuration = time.Hour
-var durationSeconds = int64(assumeRoleDefaultDuration.Seconds())
-
-type aliAuthAccount struct {
-	Account    account.Account
-	ExpiryTime *time.Time // Only set for STS accounts
+type AliAuth interface {
+	GetCredentials() (credentials.Credential, error)
+	GetOpenAPIConfig() (*openapi.Config, error)
+	GetAccount() (account.Account, error)
 }
 
-type AliAuthConfig struct {
-	account  *aliAuthAccount
-	regionID string
+type aliAuth struct {
+	accessKeyId     string
+	accessKeySecret string
+	regionId        string
+	ramRoleARN      string // optional (e,g. acs:ram::5123xxx:role/role-name)
 }
 
-func NewConfig(ramUserAccessKeyID, ramUserAccessKeySecret, regionID, ramRole, roleSessionName string) (*AliAuthConfig, error) {
-	if ramUserAccessKeyID == "" || ramUserAccessKeySecret == "" || regionID == "" {
-		return nil, fmt.Errorf("access key ID, secret, and region ID are required")
+func NewConfig(accessKeyId, accessKeySecret, regionID string, opts ...AliAuthOption) (AliAuth, error) {
```

Comment:
can we simplify the usage of this pkg by directly create return `*account.Account`?

```suggestion
func New(accessKeyId, accessKeySecret, regionID string, opts ...AliAuthOption) (*account.Account, error) {
```

Reply:
I think it's not pak, **alicatalogapis** and **OSS** need the raw STS token which is not obtainable from *account.Account

Reply:
<img width="321" alt="image" src="https://github.com/user-attachments/assets/433f68ca-78dd-4ba9-8a66-e6ceaf975ea9" />

please check if this would work or not

Reply:
Oh yes it's there
<img width="889" alt="image" src="https://github.com/user-attachments/assets/b5361ed6-757b-4ec9-b4f1-0899b9428b5d" />

But odps using this one:
<img width="893" alt="image" src="https://github.com/user-attachments/assets/10e2f901-1bfe-4fd7-9e7d-f444f82f4226" />

Should we just keep current implementation pak? Or should we just return *account.Account? 
I mean, if we remove the logic mapping ali config the package only contians these pak 😅
<img width="899" alt="image" src="https://github.com/user-attachments/assets/e58de5c0-80d8-4ab7-94da-a4dad78732d8" />

Reply:
resolved: [62abb5c](https://github.com/goto/guardian/pull/210/commits/62abb5cf5322016da076824e95badbaddc4315a1)